src/operators/map.ts

```diff
@@ -41,17 +41,14 @@ export function map<T, R>(project: (value: T, index: number) => R, thisArg?: any
     if (typeof project !== 'function') {
       throw new TypeError('argument is not a function. Are you looking for `mapTo()`?');
     }
-    return source.lift(new MapOperator(project, thisArg));
+    return source.lift(mapOperator(project, thisArg));
   };
 }
 
-export class MapOperator<T, R> implements Operator<T, R> {
-  constructor(private project: (value: T, index: number) => R, private thisArg: any) {
-  }
-
-  call(subscriber: Subscriber<R>, source: any): any {
-    return source.subscribe(new MapSubscriber(subscriber, this.project, this.thisArg));
-  }
+export function mapOperator<T, R> (project: (value: T, index: number) => R, thisArg?: any) {
```

Comment:
how much difference it'll be between `function` to property based, such as
````
const mapOperator = (project, thisArg) => (this, source) => source.subscribe(new MapSubscribe())
```?

I usually prefer it's shorthanded as long as it performs not huge different, prolly most difference is between arrow fn to normal.

Reply:
There isn't much difference, however, since we have to use the `this` context here, I purposely avoided arrow functions.

Generally, I like arrow functions better also.